### DISCUSSION TITLE
Remove `reloadCatalog` ability from `productcatalogservice`

### DISCUSSION THREAD

**Discussion by mic-max** [open-telemetry/opentelemetry-demo#288] [Ideas]
Code shown here
https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/productcatalogservice/server.go#L181

When the `reloadCatalog` bool is true the catalog file `products.json` will be read and unmarshalled and saved into the global variable `cat`. The `reloadCatalog` bool is set to True and False by sending the signals `SIGUSR1` and `SIGUSR2` respectively. This seems like some added complexity to the demo that doesn't add much value.

Aside: The catalog is also being parsed on each request, `parseCatalog` is called. This could be improved. I will work on simplifying this since this service will remain in Go lang.

**Comment by julianocosta89** [comment]
hey @mic-max do we still need this discussion?

I mean, I couldn't find `reloadCatalog` nor `parseCatalog` within https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/productcatalogservice/main.go

**Reply by mic-max** [reply]
no, these were removed in a pr of mine last week
